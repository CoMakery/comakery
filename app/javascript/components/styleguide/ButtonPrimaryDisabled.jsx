import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Button from './Button'

class ButtonPrimaryDisabled extends React.Component {
  render() {
    const {className, ...other} = this.props

    const classnames = classNames(
      'button__primary__disabled',
      className
    )

    return (
      <React.Fragment>
        <Button className={classnames} disabled {...other} />
      </React.Fragment>
    )
  }
}

ButtonPrimaryDisabled.propTypes = {
  className: PropTypes.string
}
ButtonPrimaryDisabled.defaultProps = {
  className: ''
}
export default ButtonPrimaryDisabled
