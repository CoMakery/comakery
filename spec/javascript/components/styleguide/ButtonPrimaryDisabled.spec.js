import React from 'react'
import { shallow, mount, render } from 'enzyme'
import ButtonPrimaryDisabled from 'components/styleguide/ButtonPrimaryDisabled'

describe('ButtonPrimaryDisabled', () => {
    it('renders correctly without props', () => {
      const wrapper = shallow(<ButtonPrimaryDisabled/>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.exists('.button__primary__disabled')).toBe(true)
      expect(wrapper.find('.button__primary__disabled').props().disabled).toBe(true)
    })

    it('renders correctly with custom className', () => {
      const wrapper = shallow(<ButtonPrimaryDisabled className='__test' />)

      expect(wrapper.exists('.button__primary__disabled.__test')).toBe(true)
    })
})
