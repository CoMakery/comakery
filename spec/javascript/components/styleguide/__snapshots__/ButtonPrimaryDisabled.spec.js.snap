// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimaryDisabled renders correctly without props 1`] = `
<Fragment>
  <Button
    className="button__primary__disabled"
    disabled={true}
    type="button"
    value="publish"
  />
</Fragment>
`;
